### PR TITLE
Updated branch issue#398

### PR DESCRIPTION
**Pull request by Cijin**
Contains Graphql Api request to fetch Lesson Mentors based on lesson id
Following changes made:
* Created typedef for `getLessonMentors`
* Query Resolver added for the same
* Added `db` relation between `User` and `UserLessons`

** Removing the `index.tsx` file as it was generated as well **

### CODE REVIEW

File: graphql/queryResolvers/getLessonMentors.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    })
+    //rejects: checks for promise rejection
+    //which would be the case if an error was thrown in a Promise
+    await expect(getLessonMentors(null, { lessonId: '3' })).rejects.toThrow()
+  })
+})
+
+/*
+ * TODO
+ * Does it return the correct users based on id?
```
Comment (songz) [inline graphql/queryResolvers/getLessonMentors.test.js:30]: what is this for?

File: graphql/queryResolvers/getLessonMentors.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+) => {
+  const { lessonId } = args
+  try {
+    const results = await UserLesson.findAll({
+      where: { lessonId },
+      include: [{ model: User }]
+    })
+
+    return results.map((result: any) => {
+      return result.User
```
Comment (songz) [inline graphql/queryResolvers/getLessonMentors.ts:19]: does this contribute to a leak of personal info? Like... could I specify `email` and then get user's email?

Comment (songz) [reply graphql/queryResolvers/getLessonMentors.ts:19]: https://github.com/garageScript/c0d3-app/blob/master/graphql/typeDefs.ts#L115 It does. Now anyone can get anyone's email for spam, not good I think.

Comment (songz) [reply graphql/queryResolvers/getLessonMentors.ts:19]:
```
return {
  username: result.User.username
  ...
}
```

Comment (Cijin) [reply graphql/queryResolvers/getLessonMentors.ts:19]: Hmm, yes fair point. I'll return just the username.

File: graphql/queryResolvers/getLessonMentors.test.js (hunk 1 above)
Comment (Cijin) [reply graphql/queryResolvers/getLessonMentors.test.js:30]: Sorry, I forgot to remove that. :sweat_smile:

**Review by anthonykhoa — APPROVED** [review]

**Review by anthonykhoa — CHANGES_REQUESTED** [review]
looks like theres a merge conflict

**Review by anthonykhoa — APPROVED** [review]

File: graphql/queryResolvers/getLessonMentors.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  args: ArgsGetLessonMentors
+) => {
+  const { lessonId } = args
+  try {
+    const results = await UserLesson.findAll({
+      where: { lessonId },
+      include: [{ model: User }]
+    })
+
+    return results.map((result: any) => {
```
Comment (hwong0305) [inline graphql/queryResolvers/getLessonMentors.ts:18]: Please fill in the type. We'd like to avoid using `any` since it defeats the purpose of TypeScript

Comment (Cijin) [reply graphql/queryResolvers/getLessonMentors.ts:18]: Got it. :smile: Will make that change.

Comment (Cijin) [reply graphql/queryResolvers/getLessonMentors.ts:18]: @hwong0305 What type should result be?

File: graphql/index.tsx [hunk 4]
```diff
@@ -292,6 +297,8 @@ export type AcceptSubmissionMutation = { __typename?: 'Mutation' } & {
 export type AddAlertMutationVariables = Exact<{
   text: Scalars['String']
   type: Scalars['String']
+  url?: Maybe<Scalars['String']>
+  urlCaption?: Maybe<Scalars['String']>
```
Comment (songz) [inline graphql/index.tsx:301]: why dos your pr change `AddAlertMutationVariables`?

**Review by songz — APPROVED** [review]

**Pull request merged**